nim/tests/test_routes.py

```diff
@@ -0,0 +1,103 @@
+from nim.app import app
+import pytest
+import json
+
+
+@pytest.fixture
+def test_app():
+    return app.test_client()
+
+
+def test_index(test_app):
+    response = test_app.get('/', content_type='html/text')
+    assert response.status_code == 200
+
+
+def test_new_success(test_app):
+    new_game_response = test_app.post(
+        '/new',
+        data='{"min": 10, "max": 10, "piles": 4}',
+        content_type='application/json'
+    )
+    expected_json = {"state": [10, 10, 10, 10]}
+    assert json.loads(new_game_response.data) == expected_json
+
+
+def test_new_max_limit_fail(test_app):
+    new_game_response = test_app.post(
+        '/new',
+        data='{"min": 10, "max": 51, "piles": 4}',
+        content_type='application/json'
+    )
+    expected_json = {"error": ["Please use integers between 1 and 50"]}
```

Comment:
Checking the human-readable error messages isn't a good idea. Instead of just returning `error`, we should probably have a code with it, maybe `error_code`. And you'd check the value of that code here instead, that way we'll be able to change what we display to the user without breaking out tests.

Reply:
Romain's comment here goes hand in hand with the comment that I made about returning a 400.

Reply:
I'm a little wary of ONLY checking error code here and stopping there. 
Am I right to be? Is there anything we can do closer to relying on the code for pass/fail but giving some sort of "warning" if the JSON isn't what we expect? 
@dodecaphonix @romainrichard

Reply:
I'm not sure I'm following.
We need to give an id to identify the status, and a possible error message to give more details.

For example, let's say someone called the endpoint and passed a value for `min` that was greater than `max`.
We could return a `403` (`Forbidden`) error code and an error message saying that `min` needs to be lower than `max`.

Does that help?

Reply:
Ah yes, it does. We're returning an error code along with an error message. Sounds good 🙂 
But in relation to the automation alone, you want the automation to pass or fail entirely dependant on the error code given?

Reply:
What automation?

Reply:
You would be hoping to see something like this? 
```
def test_new_max_limit_fail(client_app):
    response = client_app.post(
        '/new',
        data='{"min": 10, "max": 51, "piles": 4}',
        content_type='application/json'
    )
    assert response.status_code == 400
```

@romainrichard

Reply:
(In terms of how we're checking for an error, not necessarily that this is the exact error code we wanna give in this case)

Reply:
That's good to check the status, you could go further and check the existence of an error message.
Maybe check that `new_game_response` has the `"error"` key in it and that the length of `new_game_response["error"]` is equal to 1. That will allow us to update the error message without breaking the tests, while still make sure we do send an error message.

Reply:
Why would the length of `new_game_response["error"]` be 1?

Reply:
Because in your code, `new_game_response["error"]` is a `list` with one element: the error message. Not sure why it's a `list` but that's another topic.